docs/notes/2.21.x.md

```diff
@@ -0,0 +1,152 @@
+- [`pants.backend.python.lint.bandit`](https://www.pantsbuild.org/2.21/reference/subsystems/bandit): 1.7.5
+- [`pants.backend.python.lint.black`](https://www.pantsbuild.org/2.21/reference/subsystems/black): 23.3.0
+- [`pants.backend.python.typecheck.mypy`](https://www.pantsbuild.org/2.21/reference/subsystems/mypy): 1.4.1
+- [`pants.backend.python.lint.pyupgrade`](https://www.pantsbuild.org/2.21/reference/subsystems/pyupgrade): 3.3.2
+- [`pants.backend.python.lint.yapf`](https://www.pantsbuild.org/2.21/reference/subsystems/yapf): 0.40.2
+
+The deprecation has expired for the `python_awslambda` target alias: use the [new `python_aws_lambda_function` target](https://www.pantsbuild.org/2.21/reference/targets/python_aws_lambda_function) instead.
+
+#### Shell
+
+[The `pants.backend.shell.lint.shellcheck` backend](https://www.pantsbuild.org/2.21/reference/subsystems/shellcheck) now uses [version 0.10.0 by default](https://www.pantsbuild.org/2.21/reference/subsystems/shellcheck#version).
+
+Failing `shunit2_test` and `experimental_test_shell_command` tests can now be automatically retried by setting [the `[test].attempts_default` option](https://www.pantsbuild.org/2.21/docs/shell#retries).
+
+#### Terraform
+
+The `experimental-deploy` goal now has built-in support for [the `--dry-run` option](https://www.pantsbuild.org/2.21/reference/goals/experimental-deploy#dry_run), which runs `terraform plan` instead of `terraform apply`.
+
+[The `terraform-hcl2-parser` subsystem](https://www.pantsbuild.org/2.21/reference/subsystems/terraform-hcl2-parser) now uses `python-hcl2` version 4.3.2 by default.
+
+#### Yaml
+
+[The `pants.backend.experimental.tools.yamllint` backend](https://www.pantsbuild.org/2.21/reference/subsystems/yamllint) now uses version 1.32.0 by default.
+
+### Plugin API changes
+
+Plugins that define target generators should set the `generated_target_cls` class var to the targets that they generate. This ensures that `__defaults__` set for that `generated_target_cls` class are applied to the targets generated by the plugin. ([#20649](https://github.com/pantsbuild/pants/pull/20649))
+
+[The new `pants.backend.project_info.peek.HasAdditionalTargetDataFieldSet` union](https://www.pantsbuild.org/2.21/docs/writing-plugins/the-target-api/concepts#adding-information-to-pants-peek-output) allows attaching information to targets that is shown in the `peek` goal, using `pants peek --include-additional-info ...`.
+
+[The new `pants.core.goals.resolves.ExportableTool` union](https://www.pantsbuild.org/2.21/docs/writing-plugins/common-subsystem-tasks#making-subsystems-exportable-with-their-default-lockfile) allows subsystems & tools to be exported with `pants export ...`, similar to a user resolve. ([#20730](https://github.com/pantsbuild/pants/pull/20730))
+
+The `implicit_value` option keyword argument is no longer supported. ([#20762](https://github.com/pantsbuild/pants/pull/20762))
```

Comment:
Maybe add after "keyword argument": "for option registration"?

Reply:
```suggestion
The `implicit_value` keyword argument for option registration is no longer supported. ([#20762](https://github.com/pantsbuild/pants/pull/20762))
```